### PR TITLE
Adding hover state color for visited InlineLinks

### PR DESCRIPTION
**Pull request by ashleemboyer**
## "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] This PR's link has been added to the "📐 Pull Request" field on the Asana task
- [x] This Vercel preview link has been added to the "🔍 Preview" field on the Asana task
- [x] The description template has been filled in below

---

## Relevant links

- [Preview link](https://dev-portal-git-ambadd-visited-link-hover-hashicorp.vercel.app/waypoint/docs) 🔎
- [Asana task](https://app.asana.com/0/1201010428539925/1201959792840226/f) 🎟️

## What

- Adds a color change to visited `InlineLink`s on hover

## Why

- This was recently added to the DevDot component in Figma

## How

- Just CSS. 😊

## Testing

- [ ] Go to [/waypoint/docs](https://dev-portal-git-ambadd-visited-link-hover-hashicorp.vercel.app/waypoint/docs)
- [ ] If you haven't visited any links on this page, visit one
- [ ] Go back to the page
- [ ] Hover over a visited link
- [ ] Compare the color of the hovered visited link to [the designs](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Dev-Portal?node-id=6715%3A72182)

## Anything else?

Nope

### CODE REVIEW

File: src/components/inline-link/inline-link.module.css [hunk 1]
```diff
@@ -29,5 +29,9 @@ ones set in the surrounding text.
 
   &:visited {
     color: var(--token-color-foreground-action-visited);
+
+    &:hover {
+      color: var(--token-color-palette-purple-400);
```
Comment (didoo) [inline src/components/inline-link/inline-link.module.css:34]: @ashleemboyer there is a specific token for this: `--token-color-foreground-action-hover` https://github.com/hashicorp/design-system/blob/0d950ac091cf50d6cf06ae67a4435ddf07d050b7/packages/tokens/dist/devdot/css/tokens.css#L122

Comment (heatherlarsen) [reply src/components/inline-link/inline-link.module.css:34]: Sure 👍

Comment (didoo) [reply src/components/inline-link/inline-link.module.css:34]: @ashleemboyer FYI https://github.com/hashicorp/design-system/pull/83 
I suggest to wait and use directly that token (we should be releasing it very soon)

Comment (ashleemboyer) [reply src/components/inline-link/inline-link.module.css:34]: Will do!

**Review by zchsh — APPROVED** [review]
💅 Very nice, I dig the improvement! +1 on waiting for the token, but since it's such a small change, approving regardless (and happy to re-review if you want a set of eyes after the token update)

**Pull request merged**